### ISSUE TITLE
IPythonInputSplitter deprecated

### ISSUE THREAD

**Issue by akhmerov** [jupyter/nbconvert#1388]
nbconvert script exporter uses `IPython.core.inputsplitter` [here](https://github.com/jupyter/nbconvert/blob/master/nbconvert/filters/strings.py#L202-L210), however `IPython` warns: `` DeprecationWarning: IPython.core.inputsplitter is deprecated since IPython 7 in favor of `IPython.core.inputtransformer2` ``

**Comment by MSeal** [comment]
Yes, I started the conversation about this in https://github.com/ipython/ipython/issues/12419 -- it's a complicated enough change I put it off. Would be happy to have someone who has cycles tackle it.
